src/proxpi/server.py

```diff
@@ -87,7 +118,34 @@ def list_files(package_name: str):
     except _cache.NotFound:
         flask.abort(404)
         raise
-    return flask.render_template("files.html", package_name=package_name, files=files)
+
+    if _wants_json():
+        files_data = []
+        for file in files:
+            file_data = {"filename": file.name, "url": file.url, "hashes": {}}
```

Comment:
In the template, you generate an URL like this:

```
 <a href="{{ file.name }}{% if file.fragment %}#{{ file.fragment }}{% endif %}"
```

I think instead of ``file.url``, you want ``file.name`` for the url key.

Reply:
Good spot, thanks for that. It's fixed now